Tests/SecureXPCTests/XPCEncoder/Array encoding tests.swift

```diff
@@ -0,0 +1,232 @@
+		try assert( int8s, encodesEqualTo: createXPCArray(from:  int8s, using: { xpc_int64_create(Int64($0)) }))
+		try assert(int16s, encodesEqualTo: createXPCArray(from: int16s, using: { xpc_int64_create(Int64($0)) }))
+		try assert(int32s, encodesEqualTo: createXPCArray(from: int32s, using: { xpc_int64_create(Int64($0)) }))
+		try assert(int64s, encodesEqualTo: createXPCArray(from: int64s, using: { xpc_int64_create(Int64($0)) }))
+	}
+	
+	func testEncodes_arrayOf_UnsignedIntegers_asArrayOf_XPCUInts() throws {
+		let   uints: [UInt  ] = [.min, 0, 123, .max]
+		let  uint8s: [UInt8 ] = [.min, 0, 123, .max]
+		let uint16s: [UInt16] = [.min, 0, 123, .max]
+		let uint32s: [UInt32] = [.min, 0, 123, .max]
+		let uint64s: [UInt64] = [.min, 0, 123, .max]
+		
+		try assert(  uints, encodesEqualTo: createXPCArray(from:   uints, using: { xpc_uint64_create(UInt64($0)) }))
+		try assert( uint8s, encodesEqualTo: createXPCArray(from:  uint8s, using: { xpc_uint64_create(UInt64($0)) }))
+		try assert(uint16s, encodesEqualTo: createXPCArray(from: uint16s, using: { xpc_uint64_create(UInt64($0)) }))
+		try assert(uint32s, encodesEqualTo: createXPCArray(from: uint32s, using: { xpc_uint64_create(UInt64($0)) }))
+		try assert(uint64s, encodesEqualTo: createXPCArray(from: uint64s, using: { xpc_uint64_create(UInt64($0)) }))
+	}
+	
+	// MARK: Floating point numbers
+	
+	func testEncodes_arrayOf_Floats_asArrayOf_XPCDoubles() throws {
+		let floats: [Float] = [
+			-Float.infinity,
+			-Float.greatestFiniteMagnitude,
+			-123,
+			-0.0,
+			 0.0,
+			 123,
+			 Float.greatestFiniteMagnitude,
+			 Float.infinity
+		]
+		
+		try assert(floats, encodesEqualTo: createXPCArray(from: floats, using: { xpc_double_create(Double($0)) }))
+		
+		// These don't have regular equality, so we'll check them seperately.
+		let nans = try encode([Float.nan, Float.signalingNaN])
+		XCTAssertEqual(xpc_array_get_count(nans), 2)
+		XCTAssert(xpc_array_get_double(nans, 0).isNaN)
+		XCTAssert(xpc_array_get_double(nans, 1).isNaN)
+	}
+	
+	func testEncodes_arrayOf_Doubles_asArrayOf_XPCDoubles() throws {
+		let doubles: [Double] = [
+			-Double.infinity,
+			-Double.greatestFiniteMagnitude,
+			-123,
+			-0.0,
+			 0.0,
+			 123,
+			 Double.greatestFiniteMagnitude,
+			 Double.infinity
+		]
+		
+		try assert(doubles, encodesEqualTo: createXPCArray(from: doubles, using: { xpc_double_create(Double($0)) }))
+		
+		// These don't have regular equality, so we'll check them seperately.
+		let nans = try encode([Double.nan, Double.signalingNaN])
+		XCTAssertEqual(xpc_array_get_count(nans), 2)
+		XCTAssert(xpc_array_get_double(nans, 0).isNaN)
+		XCTAssert(xpc_array_get_double(nans, 1).isNaN)
+	}
+	
+	// MARK: Misc. types
+	
+	func testEncodes_arrayOf_Bools_asArrayOf_XPCBools() throws {
+		let bools = [false, true]
+		try assert(bools, encodesEqualTo: createXPCArray(from: bools, using: { xpc_bool_create($0) }))
+	}
+	
+	func testEncodes_arrayOf_Data_asArrayOf_XPCData() throws { // 💾
```

Comment:
> (That said, does it actually end up as a string?).


It's *much* worse than that. Storing it as a string (with say, Base64 for example), would have a 1.6x overhead (5 bits of payload per 8 bit character). The current encoding has an ***8x*** overhead. It encodes one `UInt8` of payload per `UInt64` of payload.

I think we should support the native `xpc_data` type

I think you missed a section of [one of my long comments](https://github.com/trilemma-dev/SecureXPC/pull/6#issuecomment-960865476). I quoted it below

> I think this is easy enough to do that it might be worth doing.
> 
> In particular, I've noticed that `Data` is ***really*** inefficient to encode. As it is today, it bloats sizes by 8x. It tries to encode itself as a `[UInt8]`, but each byte gets widened into an 8 byte `int64` in order to be passed to `xpc_int64_create`. Luckily, there is not per-number object allocation overhead there (that would have been brutal for my use-case haha), because the only possible int64 values here would be 0-255, all of which fit into the tagged pointer representation.
> 
> You can get a rough of idea of what that might look like, here: https://github.com/amomchilov/SecureXPC/compare/add-intial-tests-for-XPCEncoder...Alex/new-scalar-support . Of course, I only changed the encoder half, and the decoder would need to match.

Reply:
Understood. If in the future custom encoding/decoding for `Data` is added then this test would make sense, but if checked in right now I think it'd be more confusing than helpful. (And unless we want to have custom support for `Date` and `UUID` cases I'd think skipped tests for those would be equally confusing.)

Reply:
Okay, I'll remove these tests and add them in a future PR where they're supported

Reply:
I opened an issue to track this: https://github.com/trilemma-dev/SecureXPC/issues/7